openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -514,36 +629,45 @@ def _process_texture(self, filepath, do_maketx, staging, linearize, force):
                     ("Paths not found on disk, "
                      "skipping hardlink: %s") % (existing,)
                 )
-
-        if do_maketx and ext != ".tx":
-            # Produce .tx file in staging if source file is not .tx
-            converted = os.path.join(staging, "resources", fname + ".tx")
-            additional_args = [
-                "--sattrib",
-                "sourceHash",
-                texture_hash
-            ]
-            if linearize:
-                self.log.info("tx: converting sRGB -> linear")
-                additional_args.extend(["--colorconvert", "sRGB", "linear"])
-
-            config_path = get_ocio_config_path("nuke-default")
-            additional_args.extend(["--colorconfig", config_path])
-            # Ensure folder exists
-            if not os.path.exists(os.path.dirname(converted)):
-                os.makedirs(os.path.dirname(converted))
-
-            self.log.info("Generating .tx file for %s .." % filepath)
-            maketx(
-                filepath,
-                converted,
-                additional_args,
-                self.log
-            )
-
-            return converted, COPY, texture_hash
-
-        return filepath, COPY, texture_hash
+        config_path = get_ocio_config_path("nuke-default")
```

Comment:
I have to say few notes regarding subprocess arguments. If it is possible (and here it is possible) arguments should be passed to subprocess functions as list of strings so you don't have to call anything like `escape_space`.

Also I agree with @BigRoy about those explicit arguments for `TextureProcessor`. If one has different arguments then other then what is the reason of having them in abstract class... They should not be prepared for specific processor before they're passed to their processing method and all of the methods should expect the same arguments.

- `colorconvert` variable should be split to `src_colorspace` and `dst_colorspace` which are passed to process method
- the same for `color_config` which is just path to config file where colorspaces can be found
- the same for `texture_hash`
- the same for `staging_dir` (handling `converted` varible probably should not happen if MakeTX is not executed?)
In that case eash processor's `process` method arguments should expect these arguments:
```
def process(
    self,
    src_filepath,
    staging_dir,
    color_config_path,
    src_colorspace,
    dst_colorspace
):
    pass
```


So `process` method of `MakeTX` can be changed to:
```
    def process(
        self,
        src_filepath,
        staging_dir,
        color_config_path,
        src_colorspace,
        dst_colorspace
    ):
        """Make `.tx` using `maketx` with some default settings.
        The settings are based on default as used in Arnold's
        txManager in the scene.
        This function requires the `maketx` executable to be
        on the `PATH`.
        Args:
            ...fill
        Returns:
            str: Output filepath/s.
        """
        from openpype.lib import get_oiio_tools_path

        maketx_path = get_oiio_tools_path("maketx")

        if not os.path.exists(maketx_path):
            print(
                "OIIO tool not found in {}".format(maketx_path))
            raise AssertionError("OIIO tool not found")

        dirpath, filename = os.path.split(src_filepath)
        fname, ext = os.path.splitext(filename)
        dst_filepath = os.path.join(staging_dir, "resources", fname + ".tx")
        if not os.path.exists(os.path.dirname(dst_filepath)):
            os.makedirs(os.path.dirname(dst_filepath))

        cmd = [
            maketx_path,
            "-v",  # verbose
            "-u",  # update mode
            # unpremultiply before conversion (recommended when alpha present)
            "--unpremult",
            "--checknan",
            # use oiio-optimized settings for tile-size, planarconfig, metadata
            "--oiio",
            "--filter", "lanczos3",
            src_filepath,
        ]

        additional_args = []
        if src_colorspace and dst_colorspace and color_config_path:
            additional_args.extend([
                "--colorconfig", color_config_path,
                "--colorconvert", src_colorspace, dst_colorspace
            ])
        cmd.extend(additional_args)
        cmd.extend(["-o", dst_filepath])

        cmd = " ".join(cmd)

        kwargs = dict(args=cmd, stderr=subprocess.STDOUT)
        if hasattr(subprocess, "CREATE_NO_WINDOW"):
            kwargs["creationflags"] = subprocess.CREATE_NO_WINDOW

        subprocess.call(**kwargs)

        return dst_filepath
```